File: Assignment-9/hotel_client.py
import xmlrpc.client

server = xmlrpc.client.ServerProxy("http://localhost:9000")

while True:
    print("\n--- Hotel Booking Menu ---")
    print("1. Book a Room")
    print("2. Cancel Booking")
    print("3. View All Bookings")
    print("4. Exit")

    choice = input("Enter your choice: ")

    if choice == '1':
        guest_name = input("Enter guest name: ")
        result = server.book_room(guest_name)
        print(result)
    elif choice == '2':
        guest_name = input("Enter guest name to cancel booking: ")
        result = server.cancel_booking(guest_name)
        print(result)
    elif choice == '3':
        bookings = server.view_bookings()
        if bookings:
            for guest, room in bookings.items():
                print(f"Guest: {guest}, Room: {room}")
        else:
            print("No current bookings.")
    elif choice == '4':
        print("Exiting the client...")
        break
    else:
        print("Invalid choice. Try again.")

File: Assignment-9/src/HotelClient.java
import java.rmi.registry.LocateRegistry;
import java.rmi.registry.Registry;
import java.util.Scanner;

public class HotelClient {
    public static void main(String[] args) {
        try {
            Registry registry = LocateRegistry.getRegistry("localhost", 5050);
            HotelBooking stub = (HotelBooking) registry.lookup("HotelBookingService");
            Scanner sc = new Scanner(System.in);
            while (true) {
                System.out.print("1=Book, 2=Cancel, 0=Exit > ");
                int choice = sc.nextInt();
                if (choice == 0) break;
                System.out.print("Guest Name: ");
                String name = sc.next();
                System.out.print("Room No: ");
                int room = sc.nextInt();
                boolean result = (choice == 1)
                    ? stub.bookRoom(name, room)
                    : stub.cancelBooking(name, room);
                System.out.println(result ? "Success" : "Failed");
            }
            sc.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}